src/lib/y2firewall/dialogs/main.rb

```diff
@@ -58,9 +59,12 @@ def run
 
         loop do
           result = super
-          break unless result == :redraw
+          swap_api if result == :swap_mode
+          break unless result == :redraw || result == :swap_mode
         end
 
+        fw.write_only
```

Comment:
Are you writing the changes *always*? What if the user uses the `Abort` button?

Reply:
Probably yes, when we moved it from the next_handler it was not checked whether the result was `:next`or not